src/libasr/pass/intrinsic_array_function_registry.h

```diff
@@ -1316,6 +1318,265 @@ namespace Cshift {
+        ASRUtils::require_impl(x.m_args[0], "`array` argument of `eoshift` "
+            "cannot be nullptr", x.base.base.loc, diagnostics);
+        ASRUtils::require_impl(x.m_args[1], "`shift` argument of `eoshift` "
+            "cannot be nullptr", x.base.base.loc, diagnostics);
+    }
+
+    static ASR::expr_t *eval_Eoshift(Allocator &al, const Location &loc,
+            ASR::ttype_t *type, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+        ASRBuilder b(al, loc);
+        if (all_args_evaluated(args) &&
+            extract_n_dims_from_ttype(expr_type(args[0])) == 1) {
+        ASR::ArrayConstant_t *arr = ASR::down_cast<ASR::ArrayConstant_t>(ASRUtils::expr_value(args[0]));
+        ASR::ttype_t* arr_type = expr_type(args[0]);
+        std::vector<ASR::expr_t *> m_eles;
+        if (is_integer(*arr_type)) {
+            for (size_t i = 0; i < (size_t) ASRUtils::get_fixed_size_of_array(arr->m_type); i++) {
+                int ele = 0;
+                if(extract_value(ASRUtils::fetch_ArrayConstant_value(al, arr, i), ele)) {
+                    m_eles.push_back(b.i_t(ele, arr_type));
+                }
+            }
+        } else if (is_real(*arr_type)) {
+            for (size_t i = 0; i < (size_t) ASRUtils::get_fixed_size_of_array(arr->m_type); i++) {
+                double ele = 0;
+                if(extract_value(ASRUtils::fetch_ArrayConstant_value(al, arr, i), ele)) {
+                    m_eles.push_back(b.f_t(ele, arr_type));
+                }
+            }
+        } else if (is_logical(*arr_type)) {
+            for (size_t i = 0; i < (size_t) ASRUtils::get_fixed_size_of_array(arr->m_type); i++) {
+                bool ele = false;
+                if(extract_value(ASRUtils::fetch_ArrayConstant_value(al, arr, i), ele)) {
+                    m_eles.push_back(b.bool_t(ele, arr_type));
+                }
+            }
+        } else if (is_character(*arr_type)) {
+            for (size_t i = 0; i < (size_t) ASRUtils::get_fixed_size_of_array(arr->m_type); i++) {
+                std::string str = "";
+                str = ASR::down_cast<ASR::StringConstant_t>(ASRUtils::fetch_ArrayConstant_value(al, arr, i))->m_s;
+                m_eles.push_back(b.StringConstant(str, arr_type));
+            }  
+        }
+        int shift = 0;
+        if (extract_value(expr_value(args[1]), shift)) {
+            if (shift < 0) {
+                shift = m_eles.size() + shift;
+            }
+            std::rotate(m_eles.begin(), m_eles.begin() + shift, m_eles.end());
+        }
+        if (extract_value(expr_value(args[1]), shift)){
+        ASR::expr_t *final_boundary = args[2];
+            if(shift > 0){
+                int i = m_eles.size() - 1;
+                for(int j = 0; j < shift; j++){
+                    m_eles[i] = final_boundary;
+                    i--;
+                }
+            }
+            else{
+                for(int j = 0; j < (-1*shift); j++){
```

Comment:
```suggestion
            else {
                for(int j = 0; j < (-1*shift); j++) {
```
These are minor changes but if handled, will lead to a clean design.